lib/kubernetes-deploy/kubernetes_resource/pod.rb

```diff
@@ -4,50 +4,38 @@ class Pod < KubernetesResource
     TIMEOUT = 10.minutes
     SUSPICIOUS_CONTAINER_STATES = %w(ImagePullBackOff RunContainerError ErrImagePull).freeze
 
-    def initialize(name:, namespace:, context:, file:, parent: nil, logger:)
+    def initialize(name:, namespace:, context:, file: nil, parent: nil, logger:, deploy_started: nil)
       @name = name
       @namespace = namespace
       @context = context
       @file = file
       @parent = parent
+      @deploy_started = deploy_started
```

Comment:
This is an interesting semantic change here; why would a `KubernetesResource` be aware of when a deploy started?